src/libasr/asr_utils.h

```diff
@@ -289,57 +245,7 @@ static inline std::string boolop_to_str(const ASR::boolopType t) {
-        case ASR::exprType::StringLen: { return ASR::down_cast<ASR::StringLen_t>(f)->m_value; }
-        case ASR::exprType::StringItem: { return ASR::down_cast<ASR::StringItem_t>(f)->m_value; }
-        case ASR::exprType::StringSection: { return ASR::down_cast<ASR::StringSection_t>(f)->m_value; }
-        case ASR::exprType::DictLen: { return ASR::down_cast<ASR::DictLen_t>(f)->m_value; }
-        case ASR::exprType::ListLen: { return ASR::down_cast<ASR::ListLen_t>(f)->m_value; }
-        case ASR::exprType::TupleLen: { return ASR::down_cast<ASR::TupleLen_t>(f)->m_value; }
-        case ASR::exprType::SetLen: { return ASR::down_cast<ASR::SetLen_t>(f)->m_value; }
-        case ASR::exprType::LogicalNot: { return ASR::down_cast<ASR::LogicalNot_t>(f)->m_value; }
-        case ASR::exprType::StringConcat: { return ASR::down_cast<ASR::StringConcat_t>(f)->m_value; }
-        case ASR::exprType::StringRepeat: { return ASR::down_cast<ASR::StringRepeat_t>(f)->m_value; }
-        case ASR::exprType::IntegerBitNot: { return ASR::down_cast<ASR::IntegerBitNot_t>(f)->m_value; }
-        case ASR::exprType::IntegerUnaryMinus: { return ASR::down_cast<ASR::IntegerUnaryMinus_t>(f)->m_value; }
-        case ASR::exprType::RealUnaryMinus: { return ASR::down_cast<ASR::RealUnaryMinus_t>(f)->m_value; }
-        case ASR::exprType::ComplexUnaryMinus: { return ASR::down_cast<ASR::ComplexUnaryMinus_t>(f)->m_value; }
-        case ASR::exprType::ComplexRe: { return ASR::down_cast<ASR::ComplexRe_t>(f)->m_value; }
-        case ASR::exprType::ComplexIm: { return ASR::down_cast<ASR::ComplexIm_t>(f)->m_value; }
-        case ASR::exprType::ListItem: { return ASR::down_cast<ASR::ListItem_t>(f)->m_value; }
-        case ASR::exprType::TupleItem: { return ASR::down_cast<ASR::TupleItem_t>(f)->m_value; }
-        case ASR::exprType::ListSection: { return ASR::down_cast<ASR::ListSection_t>(f)->m_value; }
-        case ASR::exprType::ListPop: { return ASR::down_cast<ASR::ListPop_t>(f)->m_value; }
-        case ASR::exprType::DictPop: { return ASR::down_cast<ASR::DictPop_t>(f)->m_value; }
-        case ASR::exprType::SetPop: { return ASR::down_cast<ASR::SetPop_t>(f)->m_value; }
-        case ASR::exprType::DictItem: // Drop through
-        case ASR::exprType::ArrayConstant: // Drop through
-        case ASR::exprType::IntegerConstant: // Drop through
-        case ASR::exprType::RealConstant: // Drop through
-        case ASR::exprType::ComplexConstant: // Drop through
-        case ASR::exprType::LogicalConstant: // Drop through
-        case ASR::exprType::TupleConstant: // Drop through
-        case ASR::exprType::DictConstant: // Drop through
-        case ASR::exprType::SetConstant: // Drop through
-        case ASR::exprType::ListConstant: // Drop through
-        case ASR::exprType::StringConstant:{ // For all Constants
-            return f;
-        }
-        default : throw LFortranException("Not implemented");
-    }
+    return ASR::expr_value0(f);
```

Comment:
Happy about this change. :)

Reply:
I figured you would be. :) It's been bothering me too, so I finally fixed it.